src/providers/VoiceFocusProvider/index.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState, useEffect, useMemo, useContext, createContext, useCallback } from 'react';
+import { 
+    Device,
+    VoiceFocusTransformDevice,
+    VoiceFocusDeviceTransformer, 
+    VoiceFocusSpec, 
+    VoiceFocusDeviceOptions
+} from 'amazon-chime-sdk-js';
+
+interface Props {
+    /** Determines how how you want Amazon Voice Focus to behave */
+    spec?: VoiceFocusSpec;
+    /** Configuration for additional named arguments, including `logger` and `preload` */
+    options?: VoiceFocusDeviceOptions;
+}
+
+interface VoiceFocusState {
+    supportsVoiceFocus: boolean;
+    transformToVoiceFocus: (device: Device) => Promise<Device | VoiceFocusTransformDevice>;
+}
+
+const VoiceFocusContext = createContext<VoiceFocusState | null>(null);
+
+const VoiceFocusProvider: React.FC<Props> = ({
+    spec,
+    options,
+    children
+}) => {
+    const [supportsVoiceFocus, setSupportsVoiceFocus] = useState(false);
+    const [voiceFocusDevice, setVoiceFocusDevice] = useState<VoiceFocusTransformDevice | null>(null);
+    const [voiceFocusTransformer, setVoiceFocusTransformer] = useState<VoiceFocusDeviceTransformer | null>(null);
+
+    useEffect(() => {
+        async function initVoiceFocus() {
+            try {
+                const isVoiceFocusSupported = await VoiceFocusDeviceTransformer.isSupported(spec, options);
+                if (isVoiceFocusSupported) {
+                    const transformer = await getVoiceFocusDeviceTransformer();
+                    if(transformer && transformer.isSupported()){
+                        setSupportsVoiceFocus(true);
+                        return;
+                    }
+                }
+            } catch (e) {
+                console.warn(`Amazon Voice Focus is not supported: ${e.message}`);
```

Comment:
If you're using `console` for your logging, you can get better error reporting like this:

```typescript
console.warn('Amazon Voice Focus is not supported.', e);
```